src/backend/metal/MetalBackend.mm

```diff
@@ -102,19 +118,24 @@ void Present(nxtDevice device) {
         return new RenderPass(this, builder);
     }
     SamplerBase* Device::CreateSampler(SamplerBuilder* builder) {
-        return new Sampler(this, builder);
+        return new Sampler(builder);
     }
     ShaderModuleBase* Device::CreateShaderModule(ShaderModuleBuilder* builder) {
-        return new ShaderModule(this, builder);
+        return new ShaderModule(builder);
     }
     TextureBase* Device::CreateTexture(TextureBuilder* builder) {
-        return new Texture(this, builder);
+        return new Texture(builder);
     }
     TextureViewBase* Device::CreateTextureView(TextureViewBuilder* builder) {
-        return new TextureView(this, builder);
+        return new TextureView(builder);
     }
 
     void Device::TickImpl() {
+        resourceUploader->Tick(finishedCommandSerial);
+
```

Comment:
Good catch! It got eaten by the rebase -i. I should have ran the test after it, because right now they just hang. Done